aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1114,10 +1113,16 @@ Tensor &any_out(const Tensor &self, int64_t dim, bool keepdim, Tensor &result) {
 }
 
 Tensor &amin_out(const Tensor& self, IntArrayRef dim, bool keepdim, Tensor& result) {
-  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Illegal dtype for self, and out:", self.scalar_type(), result.scalar_type());
+  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "Illegal dtype for self, and out:",
```

Comment:
We should avoid the internal name "self" (see above) and language like illegal.

This message could be, "Expected the dtypes of input and out to match, but got [input dtype] for input's dtype and [out dtype] for out's dtype."

Reply:
I used `self` because most of python notation encourages use of `self` for naming the object on the method is being called. Given that the word `self` is so prevalent in most Python code that you tend to read elsewhere, I would  argue in favour of keeping `self` for the input argument. What do you think?